nautobot/wireless/models.py

```diff
@@ -0,0 +1,257 @@
+from django.core.validators import MinValueValidator
+from django.db import models
+
+from nautobot.core.constants import CHARFIELD_MAX_LENGTH
+from nautobot.core.models import BaseModel
+from nautobot.core.models.fields import JSONArrayField
+from nautobot.core.models.generics import PrimaryModel
+from nautobot.extras.utils import extras_features
+from nautobot.wireless.choices import (
+    RadioFrequencyChoices,
+    RadioStandardChoices,
+    WirelessAuthTypeChoices,
+    WirelessDeploymentModeChoices,
+)
+
+
+@extras_features(
+    "custom_links",
+    "custom_validators",
+    "export_templates",
+    "graphql",
+    "webhooks",
+)
+class AccessPointGroup(PrimaryModel):
+    """
+    An AccessPointGroup is a collection of AccessPoints. It is used to apply common configuration to multiple
+    AccessPoints at once.
+    """
+
+    name = models.CharField(max_length=CHARFIELD_MAX_LENGTH, unique=True)
+    description = models.CharField(max_length=200, blank=True)
+    controller = models.ForeignKey(
+        to="dcim.Controller",
+        on_delete=models.PROTECT,
+        related_name="access_point_groups",
+        blank=True,
+        null=True,
+    )
+
+    class Meta:
+        ordering = ["name"]
+
+    def __str__(self):
+        return self.name
+
+
+@extras_features(
+    "custom_links",
+    "custom_validators",
+    "export_templates",
+    "graphql",
+    "webhooks",
+)
+class DataRate(PrimaryModel):
+    """
+    A DataRates represents a data rate that can be used by a Radio.
+    """
+
+    standard = models.CharField(
+        max_length=CHARFIELD_MAX_LENGTH, choices=RadioStandardChoices, default=RadioStandardChoices.A
+    )
+    rate = models.FloatField(
+        validators=[MinValueValidator(1)],
+        help_text="Enter rate in Mbps.",
+    )
+    mcs_index = models.IntegerField(blank=True, null=True)
+
+    class Meta:
+        ordering = ["standard", "rate"]
+        unique_together = ["standard", "rate"]
+
+    def __str__(self):
+        return f"{self.standard}: {self.rate}"
+
+
+@extras_features(
+    "custom_links",
+    "custom_validators",
+    "export_templates",
+    "graphql",
+    "webhooks",
+)
+class RadioProfile(PrimaryModel):
+    """
+    A RadioProfile is a collection of settings that can be applied to a Radio.
+    """
+
+    name = models.CharField(max_length=CHARFIELD_MAX_LENGTH, unique=True)
+    frequency = models.CharField(
+        max_length=CHARFIELD_MAX_LENGTH,
+        choices=RadioFrequencyChoices,
+        default=RadioFrequencyChoices.FREQUENCY_24G,
+    )
+    tx_power_min = models.IntegerField()
+    tx_power_max = models.IntegerField()
+    channel_width = JSONArrayField(base_field=models.IntegerField())
+    allowed_channel_list = JSONArrayField(base_field=models.IntegerField(), default=list)
+    supported_data_rates = models.ManyToManyField(
```

Comment:
```suggestion
    data_rates = models.ManyToManyField(
```

https://docs.nautobot.com/projects/core/en/stable/development/core/best-practices/#field-naming-in-data-models

Reply:
Architecture specifically wanted this field to be called supported_data_rates. I will update the Model name.